packages/core/scripts/gas-rebate/index.js

```diff
@@ -0,0 +1,301 @@
+const cliProgress = require("cli-progress");
+const argv = require("minimist")(process.argv.slice(), {
+  string: ["start", "end"],
+  boolean: ["reveal-only", "claim-only"]
+});
+const fs = require("fs");
+const path = require("path");
+
+const Voting = artifacts.require("Voting");
+
+const { toBN, toWei, fromWei } = web3.utils;
+const FindBlockAtTimestamp = require("../liquidity-mining/FindBlockAtTimeStamp");
+
+const CalculateRebate = async callback => {
+  try {
+    const weekNumber = 1;
+    const endDate = argv.end ? argv.end : Math.round(Date.now() / 1000 - 60 * 5); // Default: Current time minus 5 minutes
+    const startDate = argv.start ? argv.start : endDate - 60 * 60 * 24 * 3; // Default: End time - 3 days
+    let endBlock, startBlock;
+    try {
+      endBlock = await FindBlockAtTimestamp._findBlockNumberAtTimestamp(web3, Number(endDate));
+      startBlock = await FindBlockAtTimestamp._findBlockNumberAtTimestamp(web3, Number(startDate));
+    } catch (err) {
+      console.error(err);
+    }
+    const voting = await Voting.deployed();
+    console.log(`⛽️ Calculating gas rebates from block ${startBlock} until ${endBlock}`);
+
+    // Query past contract events.
+    const [committedVotes, revealedVotes, claimedRewards] = await Promise.all([
+      voting.getPastEvents("VoteCommitted", {
+        fromBlock: 0
+        // We don't specify a start date for commits because we want to make sure we can match each
+        // reveal with a commit, even if the commit was prior to `start`.
+      }),
+      voting.getPastEvents("VoteRevealed", {
+        fromBlock: startBlock,
+        toBlock: endBlock
+      }),
+      voting.getPastEvents("RewardsRetrieved", {
+        fromBlock: startBlock,
+        toBlock: endBlock
+      })
+    ]);
+
+    // TODO: Fetch gas price data
+    const SCALING_FACTOR = toBN(toWei("1"));
+    // - Get gas price for period.  This is the ETH price per unit gas, described in Gwei.
+    const _averagePriceGweiForPeriod = "90";
```

Comment:
if only the average was 90. I guess this is why the total gas rebate was only ~500 usd.

Reply:
yep, these are just example hardcoded values for now and the final script will use the Etherscan API .